rfc/rfc-80/rfc-80.md

```diff
@@ -0,0 +1,158 @@
+6. Like native bucket tables, clustering operations are not supported.
+
+### Model change
+After the column family is introduced, the storage structure of the entire Hudi bucket table changes:
+
+![bucket](bucket.png)
+
+The bucket is divided into multiple columnFamilies by column cluster. When columnFamily is 1, it will automatically degenerate into the native bucket table.
+
+![file-group](file-group.png)
+
+### Specifying column families when creating a table
+In the table creation statement, column family division is specified in the options/tblproperties attribute;
+Column family attributes are specified in key-value mode:  
+* Key is the column family name. Format: hoodie.colFamily. Column family name    naming rules specified.  
+* Value is the specific content of the column family: it consists of all the columns included in the column family plus the precombine field. Format: " col1,col2...colN; precombineCol", the column family list and the preCombine field are separated by ";"; in the column family list the columns are split by ",".  
+
+Constraints: The column family list must contain the primary key, and columns contained in different column families cannot overlap except for the primary key. The preCombie field does not need to be specified. If not specified, the primary key will be taken by default.
+
+After the table is created, the column family attributes will be persisted to hoodie's metadata for subsequent use.
+
+### Adding and deleting column families in existing table
+Use the SQL alter command to modify the column family attributes and persist it:    
+* Execute ALTER TABLE table_name SET TBLPROPERTIES ('hoodie.columnFamily.k'='a,b,c;a'); to add a new column family.  
+* Execute ALTER TABLE table_name UNSET TBLPROPERTIES('hoodie.columnFamily.k'); to delete the column family.
+
+Specific steps are as follows:
+1. Execute the ALTER command to modify the column family
+2. Verify whether the column family modified by alter is legal. Column family modification must meet the following conditions, otherwise the verification will not pass:
+    * The column family name of an existing column family cannot be modified.  
+    * Columns in other column families cannot be divided into new column families.  
+    * When creating a new column family, it must meet the format requirements from previous chapter.  
+3. Save the modified column family to the .hoodie directory.
+
+### Writing data
+The Hudi kernel divides the input data according to column families; the data belonging to a certain column family is sorted and directly written to the corresponding column family log file.
+
+![process-write](process-write.png)
+
+Specific steps:  
+1. The engine divides the written data into buckets according to hash and shuffles the data (the writing engine completes it by itself and is consistent with the current writing of the native bucket).  
+2. The Hudi kernel sorts the data to be written to each bucket by primary key.  
```

Comment:
Just curious; is this sorting coming from flink or an external sort at the write handle level.

Reply:
Sort during the writing process，  both spark and flink has it's own ExternalSorter, we can refer those ExternalSorter to finish sort。

Reply:
ok an external sort . lets add this into the RFC (and we can resolve this comment)